Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Update/Internal/FbModificationCommandBatch.cs

```diff
@@ -0,0 +1,290 @@
+                //    && !CanBeInsertedInSameStatement(_BlockInsertCommands[0], newModificationCommand))
+                //{
+                    
+                //    CachedCommandText.Append(GetBlockInsertCommandText(commandPosition));
+                //    _BlockInsertCommands.Clear();
+                //    commandPosition = 0;
+                //}
+                //else
+                //{
+                    _BlockInsertCommands.Add(newModificationCommand);
+                    LastCachedCommandIndex = commandPosition;
+               // }
+
+            }
+            else if (newModificationCommand.EntityState == EntityState.Modified)
+            {
+                if (_BlockUpdateCommands.Count > 0
+                    && !CanBeUpdateInSameStatement(_BlockUpdateCommands[0], newModificationCommand))
+                {
+                    CachedCommandText.Append(GetBlockUpdateCommandText(commandPosition));
+                    _BlockUpdateCommands.Clear();
+                }
+                _BlockUpdateCommands.Add(newModificationCommand); 
+                LastCachedCommandIndex = commandPosition;
+            }
+            else if (newModificationCommand.EntityState == EntityState.Deleted)
+            {
+                if (_BlockDeleteCommands.Count > 0
+                    && !CanBeDeleteInSameStatement(_BlockDeleteCommands[0], newModificationCommand))
+                {
+                    CachedCommandText.Append(GetBlockDeleteCommandText(commandPosition));
+                    _BlockDeleteCommands.Clear();
+                }
+                _BlockDeleteCommands.Add(newModificationCommand);
+                LastCachedCommandIndex = commandPosition;
+            }
+            else
+            {
+                CachedCommandText.Append(GetBlockInsertCommandText(commandPosition));
+                _BlockInsertCommands.Clear(); 
+                base.UpdateCachedCommandText(commandPosition);
+            }
+        }
+         
+
+
+        private static bool CanBeDeleteInSameStatement(ModificationCommand firstCommand, ModificationCommand secondCommand)
+          => string.Equals(firstCommand.TableName, secondCommand.TableName, StringComparison.Ordinal)
+             && string.Equals(firstCommand.Schema, secondCommand.Schema, StringComparison.Ordinal)
+             && firstCommand.ColumnModifications.Where(o => o.IsWrite).Select(o => o.ColumnName).SequenceEqual(
+                 secondCommand.ColumnModifications.Where(o => o.IsWrite).Select(o => o.ColumnName))
+             && firstCommand.ColumnModifications.Where(o => o.IsRead).Select(o => o.ColumnName).SequenceEqual(
+                 secondCommand.ColumnModifications.Where(o => o.IsRead).Select(o => o.ColumnName));
+         
+        private static bool CanBeUpdateInSameStatement(ModificationCommand firstCommand, ModificationCommand secondCommand)
+    => string.Equals(firstCommand.TableName, secondCommand.TableName, StringComparison.Ordinal)
```

Comment:
Formatting.

Reply:
I improved the return, on formatting could show a suggestion?

Reply:
The indentation needs to be fixed.